### ISSUE TITLE
Typo checks in core.ops

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#18803]
https://github.com/pandas-dev/pandas/blob/master/pandas/core/ops.py#L490

```
            elif (isinstance(ovalues, datetime.datetime) and
                  hasattr(ovalues, 'tzinfo')):
```

Is that `hasattr(ovalues, 'tzinfo')` supposed to be `ovalues.tzinfo is not None`?

**Comment by gfyoung** [comment]
I can't tell ATM, so change it and find out :smile:

**Comment by jreback** [comment]
do you have an example of something that hits this? I would guess that we have *other* lines of code which work-around (so this is a no-op).

**Comment by jbrockmendel** [comment]
> do you have an example of something that hits this?

`pd.date_range('2016-01-01', periods=3).to_series() - pd.Timestamp.now()` hits it.

**Comment by jbrockmendel** [comment]
Closed by #19024
